spec/mem/tree_borrows/tree.md

```diff
@@ -0,0 +1,97 @@
+# Tree Structure in Tree Borrows
+
+The core data structure of the Tree Borrows is a *Tree*. We use a tree to track reborrows in each allocation. The tree consists of several nodes, each corresponding to a borrow tag (i.e., a reborrow operation). Each node has a parent and a list of children. Additionally, we track each tag's permissions on memory locations (bytes) within the allocation.
+
+```rust
+pub struct Tree {
+    root_tag: BorTag, 
+    /// Map a borrow tag to a node
+    nodes: Map<BorTag, Node>,
+}
+
+pub struct Node {
+    /// Borrow tag of the parent node
+    parent: Option<BorTag>, 
+    /// Borrow tags of the children node
+    children: List<BorTag>,
+    /// Permission for each location
+    permissions: Map<Address, Permission>,
```

Comment:
This can't just be `Permission`, it has to be something like `(Accessed, Permission)` where
```rust
enum Accessed {
    /// This address has been accessed (read, written, or the initial implicit read upon retag)
    /// with this borrow tag.
    Yes,
    /// This address has not yet been accessed with this borrow tag. We still track how foreign
    /// accesses affect the current permission so that on the first access, we start in the right state.
    No,
}
```

Reply:
Could you please explain why this enum is necessary?